src/herder/test/HerderTests.cpp

```diff
@@ -1800,6 +1848,25 @@ TEST_CASE("values externalized out of order", "[herder][quickRestart]")
     {
         testOutOfOrder(/* partial */ true);
     }
+    SECTION("C goes back in sync and unsticks the network")
+    {
+        testOutOfOrder(/* partial */ false);
+
+        // Now that C is back in sync and triggered next ledger
```

Comment:
Do we need to reconnect here? I think there is a slight chance that by connecting `A` and `C`, `C` externalizes ledgers that `A` and `B` externalized together.

I think we should just wait a few seconds and we should see a `NOMINATION` message from `C` for that new ledger before connecting `A` to `C` (maybe add this to the end of `testOutOfOrder`).